### ISSUE TITLE
Accessibility / a11y: Select Linear Text  does not expose selection to xsel / clipboard / PRIMARY selection

### ISSUE THREAD

**Issue by tgrushka** [xournalpp/xournalpp#6215]
### Operating System

Linux

### (Linux only) Distribution

Mint 22

### (Linux only) Desktop Environment

Cinnamon

### (Linux Only) Display Server

X.Org

### Installation Method

from recent GitHub nightly

### Xournal++ Version

1.2.5+dev b81bdc5a

### libgtk Version

3.24.41

### Bug Description

I am legally blind and want to use my screen reader (which basically uses `xsel` to get the current selection and read it aloud) in Xournal++.

When I make a selection using the **Select Linear Text** tool, a popup appears with icons that are difficult to use and get in the way of using my screen reader. Other apps, such as Firefox, Terminal, and Document Viewer, allow me to make a selection unobtrusively like any other desktop unix application. Xournal++ has this popup that:
- has icons that are too small to read (although hovering does show tooltips)
- does not go away when I hit Esc / Escape key (keyboard a11y)
- clears the selection if I click outside, as I don't want any of the actions in the popup.

Thus, **the popup cannot be dismissed without choosing an action or losing the selection.**

More critically, the `PRIMARY` selection is not exposed to `xsel` as it should be, therefore I cannot access the text at all with my screen reader.

I am unable to find any options in the menu for disabling the popup on text selection.

Since this popup is an additional (and I posit, unexpected) side-effect, is it possible to allow us to disable it?

### Expected Behaviour

When I make a text selection using the **Select Linear Text** tool:
1. Commensurate with most other Linux/Unix desktop apps, selected text should be immediately available as the `PRIMARY` selection in `xsel` as soon as the text is selected and the mouse button released (the user is finished making the selection);
2. There should be no undesired side-effects such as popups when making a text selection, especially with icon-only selections, which can present major problems for both:
  a. users with legal blindness / visual impairment (myself),
  b. totally blind users, and possibly
  c. neurodivergent users.

### Steps to Reproduce

1. Open a PDF.
2. Select **Select Linear Text** in the **Tools** menu.
3. Make a text selection somewhere in the document.
4. Notice the popup of icons.
5. Open Terminal.
6. Type `xsel`.
7. Notice that the output does **not** contain the active selection in Xournal++.

**Comment by tgrushka** [comment]
As a follow-up comment, I have solved this problem in, e.g. Flutter apps (I realize, a totally different language/framework, but in principle the same), by hooking an `onSelectionChanged` or similar callback, and using APIs, etc. to set the X selection.

For example, the following could serve as "pseudo-code" for what I'm talking about (don't know if helpful or not, I mainly work in Rust / Dart these days and have dabbled in C++ but would rather not even attempt to try to come up with C++ equivalent):

```dart
import 'dart:io';

import 'package:flutter/rendering.dart';
import 'package:flutter/services.dart';

Future<void> setClipboard(final SelectedContent? selectedContent) async {
  final text = selectedContent?.plainText;
  if (text != null) {
    if (!Platform.isLinux) {
      await Clipboard.setData(ClipboardData(text: text));
      return;
    }

    try {
      final process = await Process.start('xsel', ['--input']);
      print('SET XSEL...');
      process.stdin.writeln(text);
      await process.stdin.close();
      if (exitCode == 0) {
        print('Clipboard set successfully with xsel: $text');
      } else {
        print('Failed to set clipboard with xsel. Exit code: $exitCode');
      }
    } catch (e) {
      print('Error using xsel: $e');
    }
  }
}
```

And my Rust screen reader (a11ymate, which I intend to make open source soon), uses something like the following to retrieve the X selection when triggered via a global keyboard shortcut:
```rs
fn get_text_on_x11() -> Result<String> {
    let clipboard = Clipboard::new()?;
    let primary = clipboard.load(
        clipboard.getter.atoms.primary,
        clipboard.getter.atoms.utf8_string,
        clipboard.getter.atoms.property,
        Duration::from_millis(1000),
    )?;
    let result = String::from_utf8_lossy(&primary)
        .trim_matches('\u{0}')
        .trim()
        .to_string();
    Ok(result)
}
```
